### ISSUE TITLE
Load ProviderInformation from .well-known/openid-configuration

### ISSUE THREAD

**Issue by murashov-a** [IdentityModel/IdentityModel.OidcClient#408]
How can set EndPoints in ProviderInformation without explicitly specifying it? They are provided in .well-known/openid-configuration.

Can the library take these settings from a openid-configuration endpoint?

I get an exception that endpoints are not specified if I don't explicitly specify them

![Безымянный](https://github.com/IdentityModel/IdentityModel.OidcClient/assets/33961658/bf6ed0ef-44c5-4cb9-a06d-140bb2d7bac9)

**Comment by brockallen** [comment]
> Can the library take these settings from a openid-configuration endpoint?

Yes, that's how it works by default. You just need to set the `Authority`. See the docs here: https://identitymodel.readthedocs.io/en/latest/native/automatic.html. Or have a look at some of the samples: https://github.com/IdentityModel/IdentityModel.OidcClient.Samples

I feel like there's some confusion somewhere (by one of us), so if I'm not understanding the issue, please let us know.

**Comment by murashov-a** [comment]
Thanks for answering

Yes, the problem is on my side.

oidcclient throwed an "malformed endpoint" exception, but I don't understand what's wrong and what end point it is. Perhaps you could correct the exception text by indicating what exactly the library does not like in the openid-configuration?

Perhaps you can see the error from the json screenshot (first message)

the exception is thrown when I don't specify ProviderInformation

**Comment by brockallen** [comment]
> oidcclient throwed an "malformed endpoint" exception, but I don't understand what's wrong and what end point it is. Perhaps you could correct the exception text by indicating what exactly the library does not like in the openid-configuration?

I'm not sure what the issue is, but I have a guess that something that's not JSON is returned. This is likely due to that endpoint returning HTML... perhaps due to an error page being shown? The protocol expects JSON from the endpoint.

**Comment by brockallen** [comment]
Hello -- any update on this issue?
